src/Campaigns/Repositories/CampaignRepository.php

```diff
@@ -262,6 +262,63 @@ public function delete(Campaign $campaign): bool
         return true;
     }
 
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function mergeCampaigns(Campaign $destinationCampaign, Campaign ...$campaignsToMerge): bool
+    {
+        // Make sure the destination campaign ID will not be included into $campaignsToMergeIds
+        $campaignsToMergeIds = array_column($campaignsToMerge, 'id');
+        if ($key = array_search($destinationCampaign->id, $campaignsToMergeIds)) {
+            unset($campaignsToMergeIds[$key]);
+        }
+
+        Hooks::doAction('givewp_campaigns_merging', $destinationCampaign, $campaignsToMergeIds);
+
+        DB::query('START TRANSACTION');
+
+        try {
+            // Convert $campaignsToMergeIds to string to use it in the queries
+            $campaignsToMergeIdsString = implode(', ', $campaignsToMergeIds);
+
+            // Migrate forms from campaigns to merge to the destination campaign
+            DB::query(
+                DB::prepare("UPDATE " . DB::prefix('give_campaign_forms') . " SET campaign_id = %d WHERE campaign_id IN ($campaignsToMergeIdsString)",
+                    [
+                        $destinationCampaign->id,
+                    ])
+            );
+
+            // Migrate revenue entries from campaigns to merge to the destination campaign
+            DB::query(
+                DB::prepare("UPDATE " . DB::prefix('give_revenue') . " SET campaign_id = %d WHERE campaign_id IN ($campaignsToMergeIdsString)",
+                    [
+                        $destinationCampaign->id,
+                    ])
+            );
+
+            // Delete campaigns to merge now that we already migrated the necessary data to the destination campaign
+            DB::query("DELETE FROM " . DB::prefix('give_campaigns') . " WHERE id IN ($campaignsToMergeIdsString)");
```

Comment:
@glaubersilva since every campaign has a default form, we should also delete these references in the `give_campaign_forms` table. But that begs the question, what happens with the forms from other campaigns after the merge?

Reply:
Nice catch! I think we should keep the default form from the destination campaign because in the UI we have a text which says that all settings from the destination campaign will be kept - as a note, we'll have a new field in the settings tab to set the default form. Resolved in 446f16fbb60712c59312a1b816cf2e5f9ba8ef72

But your question makes me think that we probably should refactor the `CampaignRepository::delete()` method to also delete the entries (that are associated with the campaign being deleted) on the `give_revenue` and `give_campaign_forms` tables.

Reply:
@alaca @kjohnson Let me know what you guys think about the change related to the `CampaignRepository::delete()` method that I mentioned in my previous comment. I think we probably should handle it in a subsequent PR, but I would like to know your thoughts about it.

Reply:
@glaubersilva we shouldn't delete anything from the `give_revenue` table. 

If we keep relationships in `give_campaign_forms`, those forms will reference the campaign that no longer exists. Since forms _can't exist_ outside the context of a campaign, it will make sense to delete them. 

If we want to keep the settings data, then we can't delete anything and we need something else.

Reply:
To confirm, yes: Only the merged campaigns themselves should be deleted; all associated data (donations, revenue, etc.) should be re-associated with the final campaign being merged into.

Reply:
Go ahead and merge it 👍 

Good job, @glaubersilva